atlasdb-coordination-impl/src/main/java/com/palantir/atlasdb/coordination/keyvalue/KeyValueServiceCoordinationStore.java

```diff
@@ -325,4 +334,15 @@ private Cell getCellForSequence(long sequenceNumber) {
         return Optional.ofNullable(kvs.get(AtlasDbConstants.COORDINATION_TABLE, ImmutableMap.of(cell, Long.MAX_VALUE))
                 .get(cell));
     }
+
+    private CheckAndSetResult<SequenceAndBound> getMostRecentValueAndBound(Optional<SequenceAndBound> oldValue,
```

Comment:
nit: `getFailedResultWithMostRecentValueAndBound`? This also constructs the `CheckAndSetResult`